### ISSUE TITLE
Footer link to text (Theming App)

### ISSUE THREAD

**Issue by hume1991** [nextcloud/server#10024]
Hello, 

I use the Theming App Version: 1.4.5. I would like to display the footer link as text. Which file has to be edited? Link to the cloud: https://www.cloud.geonativ-energie.de

Best regards

Christian Huml

**Comment by MorrisJobke** [comment]
I guess there is no way to disable it as of now. IMO we should also not add yet another option here. Just set it to the URL of your instance.

Or what is the motivation of this request?

cc @nextcloud/theming @jancborchardt

**Comment by juliushaertl** [comment]
Yes, we should not introduce another option here. You could also easily hide the link with the theming_customcss app.

**Comment by hume1991** [comment]
I would have thought it is open source? I can not make any customization on my own server? I am very well good in programming and HTML and CSS. That's not a solution for me. If it continues like this, I will set up a cloud myself.

Best regards 

Christian Huml

**Comment by juliushaertl** [comment]
@hume1991 As @MariusBluem said, there is nothing limiting you from changing things on your server. Since the GitHub issue tracker is used to track enhancement requests and bug reports, we discuss here how and if requests can be integrated. You might want to checkout https://help.nextcloud.com for questions regarding customization or other questions that are not bug reports or feature requests. 

Besides that, `core/templates/layout.guest.php` might be the file you are looking for, but custom changes will of course be overwritten by every update.

**Comment by MorrisJobke** [comment]
> I would have thought it is open source? I can not make any customization on my own server? I am very well good in programming and HTML and CSS. That's not a solution for me. If it continues like this, I will set up a cloud myself.

Keep also in mind that open source does not mean, that every feature that can be imagined will be implemented. We think about it and then decide if it should be in the product or not. Keep also in mind that once it is in we as maintainers also need to take care of it for the next years and keep it working. Thus we decide carefully what to add and what not.

**Comment by hume1991** [comment]
Can I submit a request for the code of the theming app? Would expand it.

Best regards 

Christian Huml

**Comment by juliushaertl** [comment]
@hume1991 Sure any contribution is welcome there. But as said above, there should not be another option for that, since the theming app settings are already quite long and complex and adding another checkbox for a setting that has a very small target group of users conflicts with our goal to do things automatically instead of offering configuration options. (See our [design guidedlines](https://nextcloud.com/design/))

If you can come up with a different solution, feel free to propose it for discussion here. :+1:

**Comment by MorrisJobke** [comment]
> Can I submit a request for the code of the theming app? Would expand it.

As @juliushaertl pointed out: let's discuss first the possible solutions here. For this it would be super helpful if you explain the reason for this option. Maybe we can come up with a better idea to fix this problem.

**Comment by MorrisJobke** [comment]
> I see something critically opposed to links. Since with us that German law is authoritative, it can possibly lead to warnings. Besides, it can be irritating for the users. If necessary I have to develop my own app that uses the parameters of the theming app.

Could you rephrase this, because I don't get what a link has to do with German law, warnings and the need to avoid links?

**Comment by hume1991** [comment]
@MorrisJobke 

It is about the disclaimer. Each additional link has a certain liability.

Best regards 

Christian Huml

**Comment by jancborchardt** [comment]
How about: If a link isn't set in the theming app, simply don't linkify the name in the footer?
@juliushaertl?

**Comment by juliushaertl** [comment]
> How about: If a link isn't set in the theming app, simply don't linkify the name in the footer?

Yes that should work fine. Good idea @jancborchardt 

> It is about the disclaimer. Each additional link has a certain liability.

I still don't see how a link to the page itself could have influence the liability there, but if we go with the approach @jancborchardt proposed, it should be fine, right @hume1991 ?

We then need to adjust the footer generation here: https://github.com/nextcloud/server/blob/8aaf2c32729fcd0302f79546f396a07398f0d724/apps/theming/lib/ThemingDefaults.php#L154

Are you still interested in contributing? Feel free to open a pull request with the change. :wink:

**Comment by hume1991** [comment]
@jancborchardt @juliushaertl 

Would be in my sense. :)

Best regards

Christian Huml

**Comment by jancborchardt** [comment]
@hume1991 cool! Then as @juliushaertl said:

> Are you still interested in contributing? Feel free to open a pull request with the change. :wink: 

Thank you! :)

**Comment by juliushaertl** [comment]
@hume1991 You should be able to just add a check to the ThemingDefaults->getShortFooter() method, that will not return a link when the url is empty. Not sure why the other files you have listed should be affected by the change?

**Comment by hume1991** [comment]
@juliushaertl 

I would have thought the link should be officially removed from the theming app?
The variables and the text output (Theming App) would then have to be edited accordingly.

Best regards

Christian Huml

**Comment by juliushaertl** [comment]
@hume1991 Maybe you have got the discussion wrong then, the plan is to only remove the link if the link value inside the theming app is empty. If a link has been set, nothing will change.

**Comment by hume1991** [comment]
@juliushaertl 

Ok, now I understand.
Then only: server/apps/theming/lib/ThemingDefaults.php is affected.

Best regards

Christian Huml
